### ISSUE TITLE
Feature request: Expand nested unpacking and unpacking of single inline element

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#10592]
After running RUF005 in https://github.com/mhammond/pywin32/pull/2230 , I was left with some nested unpacking
![image](https://github.com/astral-sh/ruff/assets/1350584/b8d98537-e759-4105-aaa5-ed0f7a72eb31)

I feel like the following case should be detectable and flagged:
```python
# from
foo(*[bar])
foo(*(bar,))
foo(*(bar, *bar2))
foo(*(*bar2, bar, ))

# to
foo(bar)
foo(bar)
foo(bar, *bar2)
foo(*bar2, bar)
```
(same with assignments and possibly more cases I haven't thought of)

**Comment by Avasam** [comment]
Kinda looks like https://docs.astral.sh/ruff/rules/unnecessary-spread/, but for single-star unpacking `*`

**Comment by Avasam** [comment]
> The cases with sets should not be flagged, for two reasons: Elements may be deduplicated and sets are unordered.

Good point, it could only be done with literals. But "unnecessary literal collection spread" sounds like a different rule.
